fitbenchmarking/fitbenchmarking
# Table refactoring

#### Description of Work
Fixes #431  
FIxes #463

Refactor tables to use a factory classes similar to the parsers and the controllers

#### Testing Instructions

1. Check tables are the same and are understandable

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
